packages/lib-classifier/src/store/SubjectStore/SingleTextSubject/SingleTextSubject.spec.js

```diff
@@ -30,16 +61,113 @@ describe('Model > SingleTextSubject', function () {
     expect(subject.locations).to.have.lengthOf(1)
   })
 
-  describe('with an invalid subject', function () {
-    const subjectSnapshot = SubjectFactory.build({
-      locations: [
-        { 'text/plain': 'https://foo.bar/example.txt' },
-        { 'audio/mpeg': 'https://foo.bar/example.mp3' }
-      ]
+  it('should have content as expected', function () {
+    expect(subject.content).to.equal(subjectSnapshot.content)
+  })
+
+  it('should have contentLoadingState as expected', function () {
+    expect(subject.contentLoadingState).to.equal(subjectSnapshot.contentLoadingState)
+  })
+
+  describe('with multiple subject locations', function () {
+    it('should throw an error', function () {
+      expect(() => SingleTextSubject.create(multipleLocationsSubjectSnapshot)).to.throw()
     })
+  })
 
+  describe('with an invalid (not .txt) subject location', function () {
     it('should throw an error', function () {
-      expect(() => SingleTextSubject.create(subjectSnapshot)).to.throw()
+      expect(() => SingleTextSubject.create(imageSubjectSnapshot)).to.throw()
+    })
+  })
+
+  describe('with location request response that fails', function () {
+    let subject
+
+    before(async function () {
+      nock('http://localhost:8080')
+        .get('/failure.txt')
+        .reply(404)
+
+      subject = SingleTextSubject.create(failureSubjectSnapshot)
+
+      const { panoptes } = stubPanoptesJs({
+        subjects: [subject],
+        workflows: [workflowSnapshot]
+      })
+      const client = {
+        caesar: {
+          request: sinon.stub().callsFake(() => Promise.resolve({}))
+        },
+        panoptes,
+        tutorials: {
+          get: sinon.stub().callsFake(() => Promise.resolve({ body: { tutorials: [] } }))
+        }
+      }
+      const rootStore = RootStore.create({}, { client })
```

Comment:
There's a `mockStore` helper, in `@tests/mockStore`, that you can use here to quickly set up a store with workflows and subjects.

Reply:
Refactored accordingly - much cleaner, thanks for noting!